src/Common/FileCache.cpp

```diff
@@ -22,16 +22,188 @@ namespace ErrorCodes
     extern const int LOGICAL_ERROR;
 }
 
-LRUFileCache::LRUFileCache(const String & cache_base_path_, const FileCacheSettings & cache_settings_)
-    : IFileCache(cache_base_path_, cache_settings_)
+FileCache::FileCache(
+    const String & cache_base_path_,
+    const FileCacheSettings & cache_settings_)
+    : cache_base_path(cache_base_path_)
+    , max_size(cache_settings_.max_size)
+    , max_element_size(cache_settings_.max_elements)
+    , max_file_segment_size(cache_settings_.max_file_segment_size)
+    , enable_filesystem_query_cache_limit(cache_settings_.enable_filesystem_query_cache_limit)
+    , main_priority(std::make_shared<LRUFileCachePriority>())
+    , stash_priority(std::make_shared<LRUFileCachePriority>())
```

Comment:
looks like no point making main and stash priority as `shared` ptr? Let's make unique ptr or not a ptr at all.